app/scripts/migrations/118.ts

```diff
@@ -0,0 +1,78 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+import { NetworkClientId } from '@metamask/network-controller';
+import { Domain } from '@metamask/selected-network-controller';
+
+export const version = 118;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+/**
+ * Removes all Snaps domains (identified as starting with 'npm:' or 'local:') from the SelectedNetworkController's domains state.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+/**
+ * Removes all domains starting with 'npm:' or 'local:' from the SelectedNetworkController's domains state.
+ *
+ * @param state - The entire state object of the MetaMask extension.
+ */
+function transformState(state: Record<string, unknown>) {
+  const selectedNetworkControllerState = state.SelectedNetworkController;
+  if (!selectedNetworkControllerState) {
+    global.sentry?.captureException?.(
```

Comment:
We would expect this state to be missing for users updating from older versions, before we introduced this controller. I think this should be a warning rather than an error

Reply:
What does throwing a warning instead of an error in migrations look like? Not seeing any examples? Or are you saying a console warning? I'm guessing not that...

Reply:
Yep, a console warning is what I meant. Or `log.warn` at least. Here's one example: https://github.com/MetaMask/metamask-extension/blob/ea7a5374c4db8a628e348b4b3fc01f87d0211ae3/app/scripts/migrations/077.js#L45

Reply:
Oh ok. Done here: https://github.com/MetaMask/metamask-extension/pull/24409/commits/c55c61e9ebe8b7eefb873467e8b0853511caec73